DataStructures.Tests/BinarySearchTreeTests.cs

```diff
@@ -16,8 +16,8 @@ public static void Constructor_UseCustomComparer_FormsCorrectTree()
         var elems = new[] { "z", "yy", "vvv", "bbbb", "fffff", "pppppp" };
         tree.AddRange(elems);
 
-        Assert.IsNotNull(tree.Search("vvv"));
-        Assert.AreEqual("bbbb", tree.Search("vvv")!.Right!.Key);
+        Assert.That(tree.Search("vvv"), Is.Not.Null);
+        Assert.That(tree.Search("vvv")!.Right!.Key, Is.EqualTo("bbbb"));
```

Comment:
If `tree.Search("vvv")!.Right` is null, this test will fail with a null reference exception

Reply:
I added a new assert to verify if  **tree.Search("vvv")!.Right** is not null